pkg/prometheus/promcfg.go

```diff
@@ -3543,6 +3543,63 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 		})
 	}
 
+	// LinodeSDConfig
+	if len(sc.Spec.LinodeSDConfigs) > 0 {
+		configs := make([][]yaml.MapItem, len(sc.Spec.LinodeSDConfigs))
+
+		for i, config := range sc.Spec.LinodeSDConfigs {
+			s := store.ForNamespace(sc.Namespace)
+			configs[i] = cg.addSafeAuthorizationToYaml(configs[i], fmt.Sprintf("scrapeconfig/auth/%s/%s/dockersdconfig/%d", sc.GetNamespace(), sc.GetName(), i), store, config.Authorization)
+			configs[i] = cg.addOAuth2ToYaml(configs[i], s, config.OAuth2)
+			configs[i] = cg.addProxyConfigtoYaml(ctx, configs[i], sc.GetNamespace(), store, config.ProxyConfig)
+
+			if config.TLSConfig != nil {
+				configs[i] = addSafeTLStoYaml(configs[i], sc.GetNamespace(), *config.TLSConfig)
+			}
+
+			if config.Port != nil {
+				configs[i] = append(configs[i], yaml.MapItem{
+					Key:   "port",
+					Value: config.Port,
+				})
+			}
```

Comment:
(nit) please add an empty line for readability
```suggestion
			}

```